src/main/java/com/ldtteam/structurize/util/InventoryUtils.java

```diff
@@ -0,0 +1,110 @@
+package com.ldtteam.structurize.util;
+
+import com.ldtteam.structurize.api.util.ItemStackUtils;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.items.IItemHandler;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Structurize specific inventory utilities.
+ */
+public class InventoryUtils
+{
+    /**
+     * Check if an inventory has all the required stacks.
+     * @param inventory the inventory to check.
+     * @param requiredItems the list of items.
+     * @return true if so, else false.
+     */
+    public static boolean hasRequiredItems(final IItemHandler inventory, final List<ItemStack> requiredItems)
+    {
+        final List<ItemStack> listToDiscount = new ArrayList<>();
+        for (final ItemStack stack : requiredItems)
+        {
+            listToDiscount.add(stack.copy());
+        }
+
+        for (int slot = 0; slot < inventory.getSlots(); slot++)
+        {
+            final ItemStack content = inventory.getStackInSlot(slot);
+            if (content.isEmpty())
+            {
+                continue;
+            }
+            int contentCount = content.getCount();
+
+            for (final ItemStack stack : listToDiscount)
+            {
+                if (!stack.isEmpty() && stack.isItemEqual(content))
+                {
+                    if (stack.getCount() < content.getCount())
+                    {
+                        contentCount = contentCount - stack.getCount();
+                        stack.setCount(0);
+                    }
+                    else
+                    {
+                        stack.setCount(stack.getCount() - contentCount);
+                        break;
+                    }
+                }
+            }
+        }
+
+        for (final ItemStack stack : new ArrayList<>(listToDiscount))
```

Comment:
faster: loop listToDiscount and see if any stack.getCount() > 0 -> return false. After loop return true